### ISSUE TITLE
Carthage Swift Version Mismatch in Binary Downloads

### ISSUE THREAD

**Issue by Mazyod** [aws-amplify/aws-sdk-ios#1104]
**Describe the bug**
Running `carthage bootstrap --platform iOS --cache-builds` downloads the AWS frameworks binary, but rejects it due to Swift version `4.0.3` mismatch with local Swift version of `4.2.1`.

**To Reproduce**
1. Use Xcode 10.1 as the primary version on the machine
2. Run `carthage bootstrap --platform iOS --cache-builds`

**Which AWS service(s) are affected?**
Latest

**Expected behavior**
Since these are ObjC frameworks, Swift version shouldn't affect the binary downloads. If Swift is introduced to the Frameworks, it will lead to major headaches to Carthage users, since Carthage doesn't seem to handle binary downloads well with Swift versions.

**Screenshots**
```
***  Skipped installing aws-sdk-ios.framework binary due to the error:
	"Incompatible Swift version - framework was built with 4.0.3 (swiftlang-900.0.74.1 clang-900.0.39.2) and the local version is 4.2.1 (swiftlang-1000.11.42 clang-1000.11.45.1)."
```

**Environment(please complete the following information):**
 - SDK Version: 2.7.1
 - Dependency Manager: Carthage
 - Swift Version : 4.2.1

**Comment by sunchunqiang** [comment]
Hi Mazyod. 
Sorry for the inconvenience caused. This is caused by that the carthage framework is built with xcode 9.2 and we cannot generate a framework that can be compatible to all version of xcode. We are still looking for a solution for this. For now, please use  "--no-use-binaries" parameter in your carthage command.

**Comment by cornr** [comment]
Hi @Mazyod 
we have the same issue. It takes really long to build all AWS Frameworks and there is no way to only build those needed. For now our workaround is to use a 2.6.x version which afaik does not contain any Swift.
Cheers

**Comment by rohandubal** [comment]
Hello all,

With SDK release `2.7.4` we are now building binaries with Xcode version 10.1.0. This is currently the latest available Xcode version and uses Swift 4.2.1.

Please let us know if you continue to face issues.

Thanks,
Rohan
